tools/tpm2_createpolicy.c

```diff
@@ -186,48 +196,65 @@ TPM_RC build_policy(createpolicypcr_ctx *ctx,
     if (rval != TPM_RC_SUCCESS) {
         LOG_ERR("Failed Policy Get Digest\n");
         return rval;
     }
 
-    LOG_INFO("Created Policy Digest:\n");
+    printf("Policy Digest: ");
+    int i=0;
+    for(i=0; i<ctx->policy_digest.t.size; i++) {
+        printf("%02X", ctx->policy_digest.t.buffer[i] );
+    }
+    printf("\n");
+    
 
-    //save the policy buffer in a file for use later or print hex to stdout.
-    if (ctx->policy_file_flag) {
-        bool result = files_save_bytes_to_file(ctx->policyfile, (UINT8 *) &ctx->policy_digest.t.buffer,
-                ctx->policy_digest.t.size);
-        if (!result) {
-            LOG_ERR("Failed to save policy digest into file \"%s\"",
-                    ctx->policyfile);
-            return -1;
+    if(!REAL_SESSION) {
+        printf("TRIAL SESSION:\n");
+        // Get policy hash.    
+        LOG_INFO("Created Policy Digest:\n");
+    
+        //save the policy buffer in a file for use later or print hex to stdout.
+        if (ctx->policy_file_flag) {
+            bool result = files_save_bytes_to_file(ctx->policyfile, (UINT8 *) &ctx->policy_digest.t.buffer,
+                    ctx->policy_digest.t.size);
+            if (!result) {
+                LOG_ERR("Failed to save policy digest into file \"%s\"",
+                        ctx->policyfile);
+                return -1;
+            }
+        } else {
+    
+            char *s = string_bytes_to_hex(ctx->policy_digest.t.buffer, ctx->policy_digest.t.size);
+            if (!s) {
+                LOG_ERR("oom");
+                return TPM_RC_MEMORY;
+            }
+    
+            TOOL_OUTPUT(ctx->t, "policy", s);
+    
+            free(s);
         }
-    } else {
-
-        char *s = string_bytes_to_hex(ctx->policy_digest.t.buffer, ctx->policy_digest.t.size);
-        if (!s) {
-            LOG_ERR("oom");
-            return TPM_RC_MEMORY;
+    
+        // Need to flush the session here.
+        rval = Tss2_Sys_FlushContext(ctx->sapi_context,
+                ctx->policy_session->sessionHandle);
+        if (rval != TPM_RC_SUCCESS) {
+            LOG_ERR("Failed Flush Context\n");
+            return rval;
         }
-
-        TOOL_OUTPUT(ctx->t, "policy", s);
-
-        free(s);
-    }
-
-    // Need to flush the session here.
-    rval = Tss2_Sys_FlushContext(ctx->sapi_context,
-            ctx->policy_session->sessionHandle);
-    if (rval != TPM_RC_SUCCESS) {
-        LOG_ERR("Failed Flush Context\n");
-        return rval;
+    
+        // And remove the session from sessions table.
+        return tpm_session_auth_end(ctx->policy_session);
+    } else {
+        char *sh = malloc(2*sizeof(ctx->policy_session->sessionHandle));
+        sprintf(sh, "%x", ctx->policy_session->sessionHandle);
+        TOOL_OUTPUT(ctx->t, "sessionHandle", sh);
+        free(sh);
     }
-
-    // And remove the session from sessions table.
-    return tpm_session_auth_end(ctx->policy_session);
```

Comment:
I see returns that will bypass this, don't we want to make sure this is always called.